### PR TITLE
api-ts openapi-generator updates

### PR DESCRIPTION
**Pull request by manav-c**
[feat(generator): parse route operationId, platform tags, summary](https://github.com/BitGo/api-ts/commit/e06fe9476a123492306825bc97faac65783fdc34)
- Ability to parse `@operationId` jsDoc tag on `httpRoute` definitions. This populates the URL in the `dev-portal`.
- Ability to parse `@summary` jsDoc tag on `httpRoute` definitions. Previously the summary in the openAPI spec would be the name given to the `httpRoute`. This update lets you set a summary. The summary populates the LHS navigation for the route in the `dev-portal`.
- Ability to parse `@tags` jsDoc tag on `httpRoute` definitions. These are the tags that go into `platform.yaml` in the `api-docs` package and under your kit name in `dev-portal`. This allows for proper grouping/classification of your routes on the developer portal.

[feat: parse nonEmptyArrayFromQueryParam, schema for optional](https://github.com/BitGo/api-ts/commit/d95a4d738b0f5ca365b51be17e94066a059b458a)
- Ability to parse `nonEmptyArrayFromQueryParam` codec in request bodies. 
- Fixed the schema for the `optional` codec. Previously `optional` would translate to an array type which is wrong.

### CODE REVIEW

**Review by manav-c — COMMENTED** [review]
didn't realize I had to update tests. Doing that. I've used the tool locally with these updates to get BitGo Network docs published successfully.

File: packages/openapi-generator/src/expression.ts [hunk 1]
```diff
@@ -174,6 +174,21 @@ const SPECIAL_CASES: SpecialCaseMap = {
...
+        RE.map(({ schema }) => ({
+          schema: {
+            title: 'NonEmptyArrayFromQueryParam',
+            type: 'array',
+            items: schema,
+          },
+          required: true,
+        })),
+      ),
+  },
```
Comment (ericcrosson-bitgo) [inline packages/openapi-generator/src/expression.ts:191]: I don't love that this is not a conventional or public type, it's a special case that at best might be confusing to a reader not familiar with this codec or at worst, happen to match against an unintended codec (slim though the odds are! :sweat_smile:)

I don't have a suggestion to improve right now, so just creating a place for conversation and not requesting changes

**Review by ericcrosson-bitgo — COMMENTED** [review]
Thank you for working on these changes! I like the direction.

Will you please add tests to cover the new functionality you added? The tests are defined declaratively but we had to use a little jank to get them running with the expected parallelization, so let us know if you hit any trouble or have any questions.

File: packages/openapi-generator/src/expression.ts (hunk 1 above)
Comment (manav-c) [reply packages/openapi-generator/src/expression.ts:191]: That's true. The motivation behind adding this is that a lot of our routes (BitGo Network) and I think other routes within BitGo that will use `api-ts` and `io-ts` to their full extent will need to use this codec at some point. We use it for all query params in all our routes.

**Comment by ericcrosson-bitgo** [comment]
Closing for house-keeping

Please note that much of this functionality has been included in [@api-ts/openapi-generator@v2.0.0](https://github.com/BitGo/api-ts/releases/tag/%40api-ts%2Fopenapi-generator%402.0.0)

**Pull request closed without merging**